### PR TITLE
Update TNC_doc.i.in

### PR DESCRIPTION
**Pull request by mbaudin47**
Fixed typos.

### CODE REVIEW

File: python/src/TNC_doc.i.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
 %feature("docstring") OT::TNC
 "Truncated Newton Constrained solver.
 
-Tunrcated-Newton method Non-linear optimizer.
+Trunrcated-Newton method Non-linear optimizer.
```
Comment (mbaudin47) [inline python/src/TNC_doc.i.in:4]:
```suggestion
Truncated-Newton method Non-linear optimizer.
```

**Comment by jschueller** [comment]
please rebase

**Comment by mbaudin47** [comment]
> please rebase

Ok: done.

File: python/src/TNC_doc.i.in [hunk 2]
```diff
@@ -14,21 +14,21 @@ Parameters
 problem : :class:`~openturns.OptimizationProblem`
     Optimization problem to solve.
 scale : sequence of float
-    Scaling factors to apply to each variables
+    Scaling factors to apply to each variables.
```
Comment (l-brevault) [inline python/src/TNC_doc.i.in:17]:
```suggestion
    Scaling factors to apply to each variable.
```

File: python/src/TNC_doc.i.in [hunk 3]
```diff
@@ -14,21 +14,21 @@ Parameters
...
-    Constant to subtract to each variable
+    Constant to subtract to each variable.
 maxCGit : int
-    Maximum number of hessian*vector evaluation per main iteration
+    Maximum number of hessian*vector evaluation per main iteration.
 eta : float
     Severity of the line search.
 stepmx : float
-    Maximum step for the line search. may be increased during call
+    Maximum step for the line search. may be increased during call.
```
Comment (l-brevault) [inline python/src/TNC_doc.i.in:25]:
```suggestion
    Maximum step for the line search, may be increased during call.
```

**Comment by l-brevault** [comment]
Thank you for the correction of the typos in TNC documentation.
Suggestion for two minor additional corrections.

**Review by l-brevault — APPROVED** [review]

**Review by adutfoy — COMMENTED** [review]
There is still undocumented methods: set/getIgnoreFailures and get/setMaximumEvaluationNumebr: it is normal?

**Comment by mbaudin47** [comment]
> There is still undocumented methods: set/getIgnoreFailures and get/setMaximumEvaluationNumebr: it is normal?

@adutfoy : These methods were deprecated by @jschueller at 215583b3fe3582fea90c0c940dd6d4b28f69fa15. I guess that it why they are not documented anymore.

**Comment by mbaudin47** [comment]
@adutfoy : I think that this is ready. Do you agree?

**Comment by mbaudin47** [comment]
@adutfoy : I think that we are ok here: do you agree?

**Comment by mbaudin47** [comment]
@adutfoy : I think that we are ok here: do you agree?

**Review by josephmure — APPROVED** [review]

**Comment by mbaudin47** [comment]
@jschueller : I think that this PR is ready.

**Comment by adutfoy** [comment]
sorry for the delay... I approve!

**Review by adutfoy — APPROVED** [review]

**Pull request merged**